### ISSUE TITLE
Cursor position malfunctions when home and end keys are pressed.

### ISSUE THREAD

**Issue by eojina** [ueberdosis/tiptap#2704]
### What’s the bug you are facing?

bug:
When the home key or end key is pressed, the cursor moves to the beginning or end of paragraph instead of to the beginning or end of the line.

In the example at the following address, the cursor moves to the end of the paragraph rather than the end of the line.
https://codesandbox.io/s/tiptap-react-issue-template-b4hmi?file=/src/App.js

But oddly enough, the example on the tiptap official website works fine.
Official website: https://tiptap.dev/examples/default

If this was intended, it wouldn't be a bug, but I think it's not intended, it's a mistake.
Any comments would be appreciated.
I hope I am missing something.

Test Environment: Chrome 100.0.4896.88

### How can we reproduce the bug on our side?

reappearance:

```
yarn create next-app editor-test
yarn add tiptap/react tiptap/starter-kit
```

```
"dependencies": {
    "@tiptap/react": "^2.0.0-beta.108",
    "@tiptap/starter-kit": "^2.0.0-beta.183",
    "next": "12.1.4",
    "react": "18.0.0",
    "react-dom": "18.0.0"
  },
```

And paste the tiptap editor code.

```
import Tiptap from './Tiptap.jsx'

export default function Home() {
  return(
    <Tiptap />
  )
}
```

If you position the cursor in the middle of the next paragraph and press the end key, you can see that the cursor moves to the end of the paragraph, not the end of the line.

> this is a _basic_ example of **tiptap**. Sure, there are all kind of basic text styles you’d probably expect from a text editor. But wait until you see the lists:

### Can you provide a CodeSandbox?

https://codesandbox.io/s/tiptap-react-issue-template-b4hmi?file=/src/App.js

### What did you expect to happen?

When the end key is pressed, I expect to move to the end of the line rather than the end of the paragraph.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by eojina** [comment]
In the example at the following address, the cursor moves to the end of the paragraph rather than the end of the line.
https://codesandbox.io/s/tiptap-react-issue-template-b4hmi?file=/src/App.js

**Comment by bdbch** [comment]
Fixed in #2691 - will be in the next release
